core/src/main/java/com/linecorp/armeria/server/file/HttpFileServiceBuilder.java

```diff
@@ -98,11 +103,38 @@ public HttpFileServiceBuilder clock(Clock clock) {
     }
 
     /**
-     * Sets the maximum allowed number of cached file entries. If not set, {@value #DEFAULT_MAX_CACHE_ENTRIES}
-     * is used by default.
+     * Sets the maximum allowed number of cached file entries. If not set, up to {@code 1024} entries
+     * are cached by default.
      */
     public HttpFileServiceBuilder maxCacheEntries(int maxCacheEntries) {
-        this.maxCacheEntries = HttpFileServiceConfig.validateMaxCacheEntries(maxCacheEntries);
+        checkState(canSetMaxCacheEntries,
+                   "Cannot call maxCacheEntries() if called entryCacheSpec() already.");
+        entryCacheSpec = validateMaxCacheEntries(maxCacheEntries);
```

Comment:
I found out that this was a little bit weird because it validates `maxCacheEntries` but returns a cache spec. So how about just inlining this?
```java
public HttpFileServiceBuilder maxCacheEntries(int maxCacheEntries) {
    checkState(canSetMaxCacheEntries,
               "Cannot call maxCacheEntries() if called entryCacheSpec() already.");
    validateNonNegativeParameter(maxCacheEntries, "maxCacheEntries");
    if (maxCacheEntries == 0) {
        entryCacheSpec = Optional.empty();
    } else {
        // Do not have to call validateEntryCacheSpec()
        entryCacheSpec = Optional.of(String.format("maximumSize=%d", maxCacheEntries));
    }
    canSetEntryCacheSpec = false;
    return this;
}
```

Reply:
Yes, good idea. I'll fix it right away.